### PR TITLE
[runtime-security] remove userspace discarder invalidation

### PR DESCRIPTION
**Pull request by safchain**
### What does this PR do?

This PR removes the userspace discarder map operation that were done to invalidate discarder. The discarder were invalidated by the userspace to prevent a race condition when an inode discarder can be pushed after the file deletion. The PR let the kernel ebpf code handling the discarder invalidation by introduction a retention period avoiding to discard an inode. In order to remove the race between the kernel and the userspace this PR introduce a mechanism to trigger multiple map operation from the userspace using the ioctl syscall.

### Motivation

This reduce the number of syscall to invalidate discarders

### Additional Notes

This also unify the timeout mechanism for the discarders

### Describe your test plan

The kitchen test are already there to ensure there is no regression introduce by this patch.

### CODE REVIEW

File: pkg/security/ebpf/c/discarders.h [hunk 1]
```diff
@@ -39,62 +39,143 @@ int __attribute__((always_inline)) bump_discarder_revision(u32 mount_id) {
...
+    struct discarder_parameters_t *params = bpf_map_lookup_elem(discarder_map, key);
+    if (params) {
+        params->event_mask |= event_type;
+        params->timestamps[(event_type-1)&(EVENT_MAX_ROUNDED_UP-1)] = timestamp;
+    } else {
+        struct discarder_parameters_t new_params = {
+            .event_mask = event_type,
+        };
+
+        void *dst = new_params.timestamps + ((event_type-1)&(EVENT_MAX_ROUNDED_UP-1));
```
Comment (lebauce) [inline pkg/security/ebpf/c/discarders.h:66]: Why not new_params.timestamps[(event_type-1)&(EVENT_MAX_ROUNDED_UP-1)] = timestamp

File: pkg/security/ebpf/c/discarders.h [hunk 2]
```diff
@@ -39,62 +39,143 @@ int __attribute__((always_inline)) bump_discarder_revision(u32 mount_id) {
...
+    }
+
+    if (mask_has_event(params->event_mask, event_type)) {
+        return 1;
+    }
+
+    return 0;
+}
+
+// do not remove it direclty, first mark it as on hold for a period of time, after that it will be removed
```
Comment (lebauce) [inline pkg/security/ebpf/c/discarders.h:102]: nit: direclty/directly

File: pkg/security/ebpf/c/erpc.h [hunk 3]
```diff
@@ -0,0 +1,101 @@
+#ifndef _ERPC_H
+#define _ERPC_H
+
+#include "filters.h"
+
+#define RPC_CMD 0xdeadc001
+
+enum erpc_op {
+    UNKNOW_OP,
```
Comment (lebauce) [inline pkg/security/ebpf/c/erpc.h:9]: nit: UNKNOWN_OP

**Review by lebauce — APPROVED** [review]

File: pkg/security/ebpf/c/discarders.h (hunk 1 above)
Comment (safchain) [reply pkg/security/ebpf/c/discarders.h:66]: because you can't access random offset of a stack allocated struct, the verifier avoid this.

Comment (Gui774ume) [inline pkg/security/ebpf/c/discarders.h:66]: Adding `| 0` might fix the verifier bug on Centos. I don't know if the verifier is clever enough to understand what would happen if `event_type == 0`.

**Review by Gui774ume — APPROVED** [review]
lgtm

**Pull request merged**